kernel/bitpattern.h

```diff
@@ -87,6 +103,12 @@ struct BitPatternPool
 		return bits;
 	}
 
+	/**
+	 * Do two patterns match?
+	 * A pattern matches another when all their bits match.
+	 * A bit matches another if their values are equal.
+	 * A bit also matches another if either of their values is "any"
```

Comment:
Can this be summarized as "two patterns match if their intersection is non-empty"?